mobile/libuplink_android/app/src/androidTest/java/io/storj/mobile/libuplink/LibuplinkInstrumentedTest.java

```diff
@@ -0,0 +1,350 @@
+            }
+        } finally {
+            uplink.close();
+        }
+    }
+
+    @Test
+    public void testListBuckets() throws Exception {
+        Config config = new Config();
+
+        Uplink uplink = new Uplink(config, filesDir);
+        try {
+            ProjectOptions options = new ProjectOptions();
+            options.setEncryptionKey("TestEncryptionKey".getBytes());
+
+            Project project = null;
+            try {
+                project = uplink.openProject(VALID_SATELLITE_ADDRESS, VALID_API_KEY, options);
+
+                BucketConfig bucketConfig = new BucketConfig();
+                Set<String> expectedBuckets = new HashSet<>();
+                for (int i = 0; i < 10; i++) {
+                    String expectedBucket = "testBucket" + i;
+                    project.createBucket(expectedBucket, bucketConfig);
+                    expectedBuckets.add(expectedBucket);
+                }
+
+                BucketList bucketList = project.listBuckets("", 1, 100);
+                assertEquals(false, bucketList.more());
+                String aa = "";
+                for (int i = 0; i < bucketList.length(); i++) {
+                    aa += bucketList.item(i).getName() + "|";
+                }
+
+                assertEquals(aa, expectedBuckets.size(), bucketList.length());
+
+                for (String bucket : expectedBuckets) {
+                    project.deleteBucket(bucket);
+                }
+
+                bucketList = project.listBuckets("", 1, 100);
+                assertEquals(false, bucketList.more());
+                assertEquals(0, bucketList.length());
+            } finally {
+                if (project != null) {
+                    project.close();
+                }
+            }
+        } finally {
+            uplink.close();
+        }
+    }
+
+    @Test
+    public void testUploadDownloadInline() throws Exception {
+        Config config = new Config();
+
+        Uplink uplink = new Uplink(config, filesDir);
+        try {
+            ProjectOptions options = new ProjectOptions();
+            options.setEncryptionKey("TestEncryptionKey".getBytes());
+
+            Project project = null;
+            try {
+                project = uplink.openProject(VALID_SATELLITE_ADDRESS, VALID_API_KEY, options);
+
+                BucketAccess access = new BucketAccess();
+                access.setPathEncryptionKey("TestEncryptionKey".getBytes());
+
+                RedundancyScheme scheme = new RedundancyScheme();
+                scheme.setRequiredShares((short) 2);
+                scheme.setRepairShares((short) 4);
+                scheme.setOptimalShares((short) 6);
+                scheme.setTotalShares((short) 8);
+
+                BucketConfig bucketConfig = new BucketConfig();
+                bucketConfig.setRedundancyScheme(scheme);
+
+                project.createBucket("test", bucketConfig);
+
+                Bucket bucket = project.openBucket("test", access);
```

Comment:
Opened buckets should be closed. `Bucket.close()` does not do anything right now, but we should start building the habbit.